### PR TITLE
Parallelize CompactDefinition

### PR DESCRIPTION
**Pull request by bradyoo**
Needs https://github.com/Alluxio/alluxio/pull/10712 to be merged for this to not break everything.

### CODE REVIEW

**Comment by bradyoo** [comment]
@gpang

File: job/server/src/main/java/alluxio/job/plan/transform/CompactDefinition.java [hunk 1]
```diff
@@ -44,6 +45,7 @@
 public final class CompactDefinition
     extends AbstractVoidPlanDefinition<CompactConfig, ArrayList<CompactTask>> {
   private static final Logger LOG = LoggerFactory.getLogger(CompactDefinition.class);
+  private static final int JOBS_PER_WORKER = 10;
```
Comment (gpang) [inline job/server/src/main/java/alluxio/job/plan/transform/CompactDefinition.java]: Is this "job" or "task"?

File: job/server/src/main/java/alluxio/job/plan/transform/CompactDefinition.java [hunk 2]
```diff
@@ -111,7 +113,13 @@ private boolean shouldIgnore(URIStatus status) {
 
     Set<Pair<WorkerInfo, ArrayList<CompactTask>>> result = Sets.newHashSet();
     for (Map.Entry<WorkerInfo, ArrayList<CompactTask>> assignment : assignments.entrySet()) {
-      result.add(new Pair<>(assignment.getKey(), assignment.getValue()));
+      List<List<CompactTask>> partitioned = CommonUtils.partition(
+          assignment.getValue(), JOBS_PER_WORKER);
+      for (List<CompactTask> compactTasks : partitioned) {
```
Comment (gpang) [inline job/server/src/main/java/alluxio/job/plan/transform/CompactDefinition.java]: can we add a simple unit test to see that `selectExecutors()` returns multiple tasks per host?

**Comment by bradyoo** [comment]
@gpang: I added tests but as you may have noticed. This isn't so simple.

File: job/server/src/test/java/alluxio/job/plan/transform/CompactDefinitionSelectExecutorsTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+  private static final String INPUT_DIR = "/input";
+  private static final String OUTPUT_DIR = "/output";
+
+  @Test
+  public void testExecutorsParallel() throws Exception {
+    CompactConfig config = new CompactConfig(null, INPUT_DIR, OUTPUT_DIR, "test", 100);
+
+    List<URIStatus> inputFiles = new ArrayList<>();
+    for (int i = 0; i < 50; i++) {
```
Comment (gpang) [inline job/server/src/test/java/alluxio/job/plan/transform/CompactDefinitionSelectExecutorsTest.java]: NIT: can we make this a local variable like `numFiles`, and we can use it here, and also at the end of the test when we verify.

File: job/server/src/test/java/alluxio/job/plan/transform/CompactDefinitionSelectExecutorsTest.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      inputFiles.add(newFile(Integer.toString(i)));
+    }
+
+    when(mMockFileSystem.listStatus(new AlluxioURI(INPUT_DIR))).thenReturn(inputFiles);
+
+    Set<Pair<WorkerInfo, ArrayList<CompactTask>>> result = new CompactDefinition().selectExecutors(
+        config, SelectExecutorsTest.JOB_WORKERS, new SelectExecutorsContext(1,
+            new JobServerContext(mMockFileSystem, mMockFileSystemContext, mMockUfsManager)));
+    assertEquals(40, result.size());
```
Comment (gpang) [inline job/server/src/test/java/alluxio/job/plan/transform/CompactDefinitionSelectExecutorsTest.java]: Would this have been only 4 in the previous version?

Comment (bradyoo) [reply job/server/src/test/java/alluxio/job/plan/transform/CompactDefinitionSelectExecutorsTest.java]: That's correct.

**Comment by bradyoo** [comment]
feedback reflected @gpang

**Comment by gpang** [comment]
alluxio-bot, test this please

**Comment by gpang** [comment]
@bradyoo 
```
/usr/src/alluxio/job/server/src/test/java/alluxio/job/plan/transform/CompactDefinitionSelectExecutorsTest.java:43: error: Line is longer than 100 characters (found 101).
```

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Pull request merged**